fedora_messaging/cli.py

```diff
@@ -452,3 +458,41 @@ def record(exchange, queue_name, routing_key, app_name, limit, file):
     _consume(
         exchange, queue_name, routing_key, messages_recorder.collect_message, app_name
     )
+
+
+URL_TEMPLATE = "https://apps.fedoraproject.org/datagrepper/id?id={}&is_raw=true"
+
+
+@cli.command()
+@click.argument("message_id")
+@click.option("--datagrep-url", help=_datagreeper_url, default=URL_TEMPLATE)
+def replay(message_id, datagrep_url):
+    """Replay a message from Datagrepper by its message ID"""
+    message_data = _get_message(message_id, datagrep_url)
+    if message_data:
+        try:
+            _publish_message(message_data)
+            click.echo(f"Message with ID {message_id} has been successfully replayed.")
+        except Exception as e:
+            click.echo(f"Failed to replay message: {e}")
+    else:
+        click.echo(f"Could not find message with ID {message_id}")
+
+
+def _get_message(message_id, datagrep_url):
+    """Fetch a message by ID from Datagreeper"""
+    url = datagrep_url.format(message_id)
+    response = requests.get(url, timeout=5)
+    if response.ok:
+        return response.json()
+    else:
+        click.echo(
+            f"Failed to retrieve message from Datagrepper: HTTP {response.status_code}"
+        )
+        return None
```

Comment:
Instead of returning `None` you could just let the exception raised by `response.raise_for_status()` propagate upwards and catch it where it's most appropriate to handle it. Otherwise you're doing a test on `response.ok` and another one on `message_data` being `None`. Only one is necessary, this will simplify the code.